modules/stochastic_tools/src/surrogates/PolynomialChaosTrainer.C

```diff
@@ -44,53 +49,139 @@ PolynomialChaosTrainer::PolynomialChaosTrainer(const InputParameters & parameter
     paramError("distributions",
                "Sampler number of columns does not match number of inputted distributions.");
 
-  if (_quad_sampler && (!_pvals.empty() || _pcols.size() != _sampler.getNumberOfCols()))
+  auto rtype_enum = getParam<MooseEnum>("regression_type");
+  if (rtype_enum == "auto")
+    _rtype = _quad_sampler ? 0 : 1;
+  else
+    _rtype = rtype_enum == "integration" ? 0 : 1;
+
+  if (_rtype == 0 && _quad_sampler &&
+      (!_pvals.empty() || _pcols.size() != _sampler.getNumberOfCols()))
```

Comment:
I think we should add an error if the number of polynomial terms is larger than the number of samples when we choose OLS.